src/main/java/it/gov/pagopa/reward/service/synchronous/op/CreateTrxSynchronousServiceImpl.java

```diff
@@ -88,16 +91,21 @@ private Mono<Boolean> checkInitiative(SynchronousTransactionRequestDTO request,
                 .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_INITIATIVE_NOT_FOUND));
     }
 
-    private Mono<Boolean> checkOnboarded(SynchronousTransactionRequestDTO request, TransactionDTO trx, String initiativeId) {
+    private Mono<Pair<InitiativeConfig, OnboardingInfo>> checkOnboarded(SynchronousTransactionRequestDTO request, TransactionDTO trx, String initiativeId) {
         return onboardedInitiativesService.isOnboarded(trx.getHpan(), trx.getTrxChargeDate(), initiativeId)
-                .map(b -> checkingResult(b, request, initiativeId, RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE));
+                .switchIfEmpty(Mono.error(getTransactionSynchronousException(request, initiativeId, RewardConstants.TRX_REJECTION_REASON_NO_INITIATIVE)));
     }
 
     private Boolean checkingResult(Boolean b, SynchronousTransactionRequestDTO request, String initiativeId, String trxRejectionReasonNoInitiative) {
         if (b.equals(Boolean.TRUE)) {
             return Boolean.TRUE;
         } else {
-            throw new TransactionSynchronousException(syncTrxRequest2TransactionDtoMapper.apply(request, initiativeId, List.of(trxRejectionReasonNoInitiative)));
+            throw getTransactionSynchronousException(request,initiativeId, trxRejectionReasonNoInitiative);
         }
     }
+    @NotNull
+    private TransactionSynchronousException getTransactionSynchronousException(SynchronousTransactionRequestDTO request, String initiativeId, String trxRejectionReason) {
```

Comment:
why "get" you are building it, not retrieving from an external source
```suggestion
    private TransactionSynchronousException buildTransactionSynchronousException(SynchronousTransactionRequestDTO request, String initiativeId, String trxRejectionReason) {
```